File: Matérias/AEDS 2/BeeCrowd/AssuntosPendentes.c
#include <stdio.h>
#include <string.h>

int main() {
    char S[100005];
    int aberto = 0;

    fgets(S, sizeof(S), stdin);
    // Remove o '\n' do final, se houver
    S[strcspn(S, "\n")] = 0;

    for (int i = 0; S[i] != '\0'; i++) {
        if (S[i] == '(') {
            aberto++;
        } else if (S[i] == ')') {
            if (aberto > 0)
                aberto--;
        }
    }

    if (aberto > 0)
        printf("Ainda temos %d assunto(s) pendente(s)!\n", aberto);
    else
        printf("Partiu RU!\n");

    return 0;
}

File: Matérias/AEDS 2/BeeCrowd/AssuntosPendentes.java

import java.util.Scanner;

public class AssuntosPendentes {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String S = scanner.nextLine();
        int aberto = 0;
        for(int i = 0; i < S.length(); i++)
        {
            char c = S.charAt(i);
            if(c == '(')
                aberto++;
            else{
                if(aberto > 0)
                    aberto--;
            }
        }
        if(aberto > 0)
            System.out.println("Ainda temos " + aberto + " assunto(s) pendente(s)!");
        else 
            System.out.println("Partiu RU!");
    }
}
